### ISSUE TITLE
yarr is feature complete

### ISSUE THREAD

**Issue by nkanaev** [nkanaev/yarr#57]
thank you for your interest in the project. unfortunately, i cannot accept any more new features.
the project was created due to lack of the application that i wanted, and it's already serving my own needs. i'd like to keep it simple.

yarr is not being deprecated. it's my go-to feed reader, and i'll gladly fix bugs & accept patches to improve the existing functionality.

my only requests for contributors would be:

- to not introduce any 3rd party dependencies
- to cover new code with unit tests whenever possible

if you'd like to introduce new features, please do so by forking it. i won't guarantee that it'd be merged to the upstream though.

**Comment by geekedtv** [comment]
Not even a grid card view? :( Love the app though!

**Comment by nkanaev** [comment]
@geekedtv no, sorry.

**Comment by FinecoFinit** [comment]
@nkanaev maybe at least feed with big images/adjustable grid for nitter rss implementation; also, would be good to have ability to install as app on phone

**Comment by Denyerec** [comment]
Unless I'm missing some pre-requisite, downloading, extracting, and double clicking yarr.exe (win 10 64) doesn't appear to do/load anything. No error messages, nothing in the windows protected folder log, nowt. Bit lost as to how to proceed!

**Comment by Denyerec** [comment]
> @Denyerec updated the readme with further instructions.

Yep - that got buried by Windows in short order, completely missed it! Many thanks!

**Comment by ljmccarthy** [comment]
Will you be making any further bug fix releases?

**Comment by nkanaev** [comment]
> Will you be making any further bug fix releases?

Yes. The upcoming release is full of them.

**Comment by adaszko** [comment]
Hi!  I love yarr and use it all the time but there's only feature I sorely miss: automatic expiration (or marking as read) of articles after a set period.  I mean it as a digital wellbeing sort of thing.  RSS is order of magnitude better than any algorithmically-controlled news feeds like Facebook/News sites, but you can still fall into the dopamine craving trap because given enough feeds, there's always going to be something unread in your reader.  If articles were marked as read automatically after a set period, it would be beneficial to the mental wellbeing.  The reasoning is: if an article wasn't interesting enough to be read before (let's say) a week passed, there's a high chance it's simply isn't interesting, and there's no loss in expiring it.

**Comment by nkanaev** [comment]
Hi. *yarr* was designed with my 2 personal needs in mind:

1. less parameters - I don't like tinkering with settings, and prefer to have them at the bare minimum.
2. less distractions - I don't like notifications and number badges. *yarr* doesn't have former and limits the latter only to unread/starred tabs.

> ...dopamine craving trap because given enough feeds, there's always going to be something unread in your feed reader...

For me, personal restraint and curating the feeds was the only solution. But since we're talking about your needs, we could either:

1. add a setting to disable badge numbers altogether.
2. create a periodic task to mark articles as read. I'd rather not add the task as a setting, but I'm happy to provide an external cron task recipe. This could be a shell one-liner, which is either:
    * a sqlite3 command to the local db
    * an external call to the API endpoint (via curl)

What do you think?

**Comment by adaszko** [comment]
> Hi. *yarr* was designed with my 2 personal needs in mind:
> 
> 1. less parameters - I don't like tinkering with settings, and prefer to have them at the bare minimum.
> 2. less distractions - I don't like notifications and number badges. *yarr* doesn't have former and limits the latter only to unread/starred tabs.
> 
> > ...dopamine craving trap because given enough feeds, there's always going to be something unread in your feed reader...
> 
> For me, personal restraint and curating the feeds was the only solution. But since we're talking about your needs, we could either:
> 
> 1. add a setting to disable badge numbers altogether.
> 2. create a periodic task to mark articles as read. I'd rather not add the task as a setting, but I'm happy to provide an external cron task recipe. This could be a shell one-liner, which is either:
>     * a sqlite3 command to the local db
>     * an external call to the API endpoint (via curl)
> 
> What do you think?

An sqlite command would suffice.  Thank you!

**Comment by nkanaev** [comment]
> An sqlite command would suffice. Thank you!

```
sqlite3 /path/to/your/local.db "update items set status = 1 where date < date('now', '-90 days')"
```

Adjust the date (`-90 days`) as needed using the rules documented [here](https://www.sqlite.org/lang_datefunc.html#modifiers).

**Comment by fredrikfischer** [comment]
I have a small feature suggestion. To be able to update not only the feed name, but also the feed URL. I use nitter, and sometimes instances fail and the URL have to be updated (nitter.nl -> nitter.cz)

Thank you for your nice work :)

**Comment by a-bali** [comment]
Hello, can I have two feature suggestions:
1. Some feeds contain a separate link for comments of a feed entry with the <comments> tag. See https://news.ycombinator.com/rss for example. Would it be possible to place a "Open comments in browser" button on the top toolbar if such link exists? Maybe also an option to load the comments page within the window.
2. Would it be possible to make the "Read here" button "sticky" for a feed i.e. if I use it for one item, use it automatically for all other items within the feed?
Thank you!

**Comment by a-bali** [comment]
> 1. Some feeds contain a separate link for comments of a feed entry with the  tag. See https://news.ycombinator.com/rss for example. Would it be possible to place a "Open comments in browser" button on the top toolbar if such link exists? Maybe also an option to load the comments page within the window.

Meanwhile I already created a PR for this.

**Comment by sekrause** [comment]
> Some feeds contain a separate link for comments of a feed entry with the tag. See https://news.ycombinator.com/rss for example. Would it be possible to place a "Open comments in browser" button on the top toolbar if such link exists? Maybe also an option to load the comments page within the window.

@a-bali You can do something like that client-side with a simple userscript, I've posted mine for Hacker News in a new discussion: https://github.com/nkanaev/yarr/discussions/222

**Comment by yoshi314** [comment]
in case you want to reference some packaging scripts, here's one for Gentoo

https://github.com/yoshi314/yoshi314-overlay/tree/master/net-news/yarr

**Comment by nkanaev** [comment]
I'm planning to brush up the project and introduce some changes and improvements, along with some new features.
